doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -7,14 +7,62 @@ Security
    :local:
    :depth: 3
 
-Nordic Matter samples leverage security features supported in the |NCS| that can be divided into three major categories:
+Nordic Matter samples leverage security features supported in the |NCS| that can be divided into four major categories:
 
+* Trusted Execution Environment
 * Cryptography
 * Trusted storage
 * Securing production devices
 
 In the following sections you will learn more details about each listed category.
 
+Trusted Execution Environment
+*****************************
+
+In Matter samples, you can use two types of Trusted Execution Environments (TEE) to provide a secure environment for the execution of security-sensitive code, depending on the SoC:
+
+* nRF54L Series devices can use Trusted Firmware-M (TF-M).
+* nRF54H Series devices can use the Secure Domain.
+
+All cryptographic operations are performed using the PSA Cryptography API.
+
+nRF54H Secure Domain
+====================
+
+The Secure Domain is a dedicated domain that executes a pre-compiled, Nordic Semiconductor-signed firmware component.
+This firmware allows the execution of all secure operations in a separate domain, providing a secure environment for security-sensitive Matter code.
+
+In Matter samples, we use security services exposed by the Secure Domain to perform all cryptographic operations during commissioning to Matter and while operating in the Matter network.
+All keys used by Matter, such as session keys, DAC private keys, and other keys, are stored in the Secure Domain's secure space and are isolated from local domains.
+
+To learn more about the secure domain, see the :ref:`ug_nrf54h20_architecture_cpu` user guide.
```

Comment:
Do we really need to duplicate information from the nRF54H user guides? Can't we just link to it in the bullet point above and add information about Matter specifics ("In Matter samples...") in the bullet point entry?